engine/rig/src/rig.cpp

```diff
@@ -970,61 +1038,28 @@ namespace dmRig
-
-            uint32_t index_count = mesh->m_PositionIndices.m_Count;
-
-            // Bump scratch buffer capacity to handle current vertex count
-            if (positions.Capacity() < index_count) {
-                positions.OffsetCapacity(index_count - positions.Capacity());
-            }
-            positions.SetSize(index_count);
-
-            if (normals.Capacity() < index_count) {
-                normals.OffsetCapacity(index_count - normals.Capacity());
-            }
-            normals.SetSize(index_count);
+        // TODO: Currently, we only have support for a single material so we bake all meshes into one
+        uint32_t vertex_count = mesh->m_Positions.m_Count / 3;
 
-            float* positions_buffer = (float*)positions.Begin();
-            float* normals_buffer = (float*)normals.Begin();
+        // Bump scratch buffers capacity to handle current vertex count
+        EnsureSize(positions, vertex_count);
+        EnsureSize(normals, vertex_count);
+        EnsureSize(tangents, vertex_count);
 
-            // Transform the mesh data into world space
+        float* positions_buffer = (float*)positions.Begin();
+        float* normals_buffer = (float*)normals.Begin();
+        float* tangents_buffer = (float*)tangents.Begin();
 
-            dmRig::GeneratePositionData(mesh, world_matrix, pose_matrices, positions_buffer);
-            if (mesh->m_NormalsIndices.m_Count) {
-                dmRig::GenerateNormalData(mesh, normal_matrix, pose_matrices, normals_buffer);
-            }
+        // Transform the mesh data into world space
+        dmRig::GeneratePositionData(mesh, world_matrix, pose_matrices, positions_buffer);
+        if (mesh->m_Normals.m_Count) {
+            dmRig::GenerateNormalData(mesh, normal_matrix, pose_matrices, normals_buffer, tangents_buffer);
+        }
 
-            vertex_data_out = WriteVertexData(mesh, positions_buffer, normals_buffer, vertex_data_out);
-        }
-
-        // DEF-3610
-        // Using Wasm on Microsoft Edge this function returns NULL after a couple of runs.
-        // There is no code path that could result in NULL, leading us to suspect it has
-        // to do with some runtime optimization.
-        // If we add some logic that touches vertex_data_out the function keeps returning
-        // valid values, so we add an assert to verify vertex_data_out never is NULL.
-        // However this means we need assert on Emscripten, so for now we output a
-        // compile error if the engine is built with NDEBUG.
-#ifdef __EMSCRIPTEN__
-#ifdef NDEBUG
-        #error "DEF-3610 - Can't compile with NDEBUG since dmRig::GenerateVertexData currently depends on assert() to work on Emscripten builds."
-#else
-        assert(vertex_data_out != 0x0);
-#endif
-#endif
-
-        return vertex_data_out;
+        return WriteVertexData(mesh, positions_buffer, normals_buffer, tangents_buffer, vertex_data_out);
```

Comment:
We _really_ want skinning on the GPU so we can deprecate this CPU code path.

Reply:
+inf on that.. ;(